packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -154,16 +168,23 @@ export class CheckboxBase extends React.Component<ICheckboxProps, ICheckboxState
   };
 
   private _onChange = (ev: React.FormEvent<HTMLElement>): void => {
-    const { disabled, onChange } = this.props;
-    const { isChecked } = this.state;
+    const { disabled, onChange, removeIndeterminate, indeterminate, defaultIndeterminate } = this.props;
+    const { isChecked, isIndeterminate } = this.state;
 
     if (!disabled) {
-      if (onChange) {
-        onChange(ev, !isChecked);
-      }
-
-      if (this.props.checked === undefined) {
-        this.setState({ isChecked: !isChecked });
+      if (!isIndeterminate) {
+        if (onChange) {
+          onChange(ev, !isChecked);
+        }
+        if (this.props.checked === undefined) {
+          this.setState({ isChecked: !isChecked });
+        }
+      } else {
+        if (defaultIndeterminate) {
+          this.setState({ isIndeterminate: false });
+        } else if (removeIndeterminate && indeterminate) {
+          removeIndeterminate();
```

Comment:
`removeIndeterminate` should be called regardless of whether it's controlled or not, to match the pattern with `onChange`

Reply:
Sounds good, switched it over to match `onChange`